src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -361,14 +569,30 @@ void EIRPlantLoopHeatPump::doPhysics(EnergyPlusData &state, Real64 currentLoad)
     }
     Real64 const sourceMCp = this->sourceSideMassFlowRate * CpSrc;
     this->sourceSideOutletTemp = this->calcSourceOutletTemp(this->sourceSideInletTemp, this->sourceSideHeatTransfer / sourceMCp);
+
+    if (this->waterSource && abs(this->sourceSideOutletTemp - this->sourceSideInletTemp) > 100.0) { // whoaa out of range happenings on water loop
+        //
+        // TODO setup recurring error warning?
+        // lets do something different than fatal the simulation
+        if ((this->sourceSideMassFlowRate / this->sourceSideDesignMassFlowRate) < 0.01) { // current source side flow is 1% of design max
+            // just send it all to skin losses and leave the fluid temperature alone
+            this->sourceSideOutletTemp = this->sourceSideInletTemp; // TRANE
+        } else if (this->sourceSideOutletTemp > this->sourceSideInletTemp) {
+            this->sourceSideOutletTemp = this->sourceSideInletTemp + 100.0; // cap it at 100C delta
+
+        } else if (this->sourceSideOutletTemp < this->sourceSideInletTemp) {
+            this->sourceSideOutletTemp = this->sourceSideInletTemp - 100.0; // cap it at 100C delta
+        }
+    }
 }
 
 void EIRPlantLoopHeatPump::onInitLoopEquip(EnergyPlusData &state, [[maybe_unused]] const PlantLocation &calledFromLocation)
 {
     // This function does all one-time and begin-environment initialization
     std::string static const routineName = std::string("EIRPlantLoopHeatPump :") + __FUNCTION__;
 
-    this->oneTimeInit(state); // plant setup
+    this->oneTimeInit(state);          // plant setup
+    this->isPlantInletOrOutlet(state); // check location
 
     if (state.dataGlobal->BeginEnvrnFlag && this->envrnInit && state.dataPlnt->PlantFirstSizesOkayToFinalize) {
```

Comment:
Though not really part of this PR, find that this routine is not coded properly for iterative plant sizing and HVAC Sizing Simulations.  Plant components should continually resize during a set of plant sizing iterations and then report when finalized.  Putting the sizing calls behind this if statement is not going to work properly because it would size during early iterations (PlantFirstSizesOkayToFinalize = false) nor will it reenter multiple times once this->envrnInit gets set to false.  

Testing with coincident plant sizing using HVAC Sizing Simulations showed that final sizes are not reported, only initial sizes.